src/protocol/DataFeed.sol

```diff
@@ -38,6 +38,17 @@ contract DataFeed is IDataFeed {
         require(numBlobs > 0, "no data to publish");
 
         uint256 nQueries = queries.length;
+
+        uint256 totalValue;
+        bytes[] memory metadata = new bytes[](nQueries);
+        for (uint256 i; i < nQueries; ++i) {
+            metadata[i] = IPublicationHook(queries[i].provider).beforePublish{value: queries[i].value}(
+                msg.sender, queries[i].input
+            );
+            totalValue += queries[i].value;
```

Comment:
beforePublish and postPublish may use different value...

Reply:
can we just avoid sending ETH in the post-publish case? Any ETH it needs can be passed along with the pre-publish hook, and the hook contract can partition the value if necessary.

Reply:
I agree that having different pre-post hook queries would be cleaner.